src/Hosting/Hosting/src/Internal/HostingApplication.cs

```diff
@@ -122,13 +122,26 @@ internal class Context
         {
             public HttpContext? HttpContext { get; set; }
             public IDisposable? Scope { get; set; }
-            public Activity? Activity { get; set; }
+            public Activity? Activity
+            {
+                get => HttpActivityFeature?.Activity;
+                set
+                {
+                    if (HttpActivityFeature is null)
+                    {
+                        HttpActivityFeature = new ActivityFeature(value!);
+                    }
+                    HttpActivityFeature.Activity = value!;
+                }
+            }
             internal HostingRequestStartingLog? StartLog { get; set; }
 
             public long StartTimestamp { get; set; }
             internal bool HasDiagnosticListener { get; set; }
             public bool EventLogEnabled { get; set; }
 
+            internal IHttpActivityFeature? HttpActivityFeature;
```

Comment:
I think this is what we were talking about?

Reply:
Yep

Reply:
Still fast to get and set with `IFeatureCollection`?

Reply:
It ends up being a field in Kestrel.

Reply:
But it is still set into `IFeatureCollection`. The code for IActivityFeatre as a known feature was removed so it is added into the ExtraFeatures dictionary. There will be overhead in getting and setting it into the dictionary.

Reply:
It will be added as a known feature.

Reply:
What is a known feature? Common, Sometimes, or Maybe?

Reply:
always